### PR TITLE
dom/met signature: `make_cast_default`, `make_is_equal`

### LINKED ISSUE #722: partial `make_cast`

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #722 

Updates the following constructors to partial format:
- `make_cast_default`
- `make_df_cast_default`
- `make_is_equal`
- `make_df_is_equal`

TODO: drop changes to `df_` constructors.

### CODE REVIEW

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #648** <a href="https://app.graphite.dev/github/pr/opendp/opendp/648" target="_blank">Graphite</a> 
    * **PR #618** <a href="https://app.graphite.dev/github/pr/opendp/opendp/618" target="_blank">Graphite</a> 
      * **PR #674** <a href="https://app.graphite.dev/github/pr/opendp/opendp/674" target="_blank">Graphite</a> 
        * **PR #675** <a href="https://app.graphite.dev/github/pr/opendp/opendp/675" target="_blank">Graphite</a> 
          * **PR #676** <a href="https://app.graphite.dev/github/pr/opendp/opendp/676" target="_blank">Graphite</a> 
            * **PR #678** <a href="https://app.graphite.dev/github/pr/opendp/opendp/678" target="_blank">Graphite</a> 
              * **PR #682** <a href="https://app.graphite.dev/github/pr/opendp/opendp/682" target="_blank">Graphite</a> 
                * **PR #684** <a href="https://app.graphite.dev/github/pr/opendp/opendp/684" target="_blank">Graphite</a> 
      * **PR #645** <a href="https://app.graphite.dev/github/pr/opendp/opendp/645" target="_blank">Graphite</a> 
        * **PR #604** <a href="https://app.graphite.dev/github/pr/opendp/opendp/604" target="_blank">Graphite</a> 
          * **PR #687** <a href="https://app.graphite.dev/github/pr/opendp/opendp/687" target="_blank">Graphite</a> 
            * **PR #688** <a href="https://app.graphite.dev/github/pr/opendp/opendp/688" target="_blank">Graphite</a> 
              * **PR #689** <a href="https://app.graphite.dev/github/pr/opendp/opendp/689" target="_blank">Graphite</a> 
                * **PR #690** <a href="https://app.graphite.dev/github/pr/opendp/opendp/690" target="_blank">Graphite</a> 
                  * **PR #512** <a href="https://app.graphite.dev/github/pr/opendp/opendp/512" target="_blank">Graphite</a> 
                  * **PR #692** <a href="https://app.graphite.dev/github/pr/opendp/opendp/692" target="_blank">Graphite</a>  👈
        * **PR #691** <a href="https://app.graphite.dev/github/pr/opendp/opendp/691" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/692?utm_source=stack-comment).

File: rust/src/transformations/manipulation/ffi.rs [hunk 1]
```diff
@@ -180,8 +200,11 @@ mod tests {
     #[test]
     fn test_make_is_equal() -> Fallible<()> {
         let transformation = Result::from(opendp_transformations__make_is_equal(
+            util::into_raw(AnyDomain::new(VectorDomain::new(
+                AtomDomain::<i32>::default(),
+            ))),
+            util::into_raw(AnyMetric::new(SymmetricDistance::default())),
```
Comment (andrewvyrros) [inline rust/src/transformations/manipulation/ffi.rs:206]: Can you use `AnyDomain::new_raw`/`AnyMetric::new_raw` here?

**Review by andrewvyrros — COMMENTED** [review]
Getting there! A couple of tiny issues, and `partial_eq` accidentally got renamed to `part_eq`.

File: rust/src/ffi/any.rs [hunk 2]
```diff
@@ -41,13 +41,13 @@ impl<const CLONE: bool, const PARTIALEQ: bool, const DEBUG: bool>
     fn new_base<T: 'static>(
         value: T,
         clone_glue: Option<Glue<fn(&Self) -> Self>>,
-        partial_eq_glue: Option<Glue<fn(&Self, &Self) -> bool>>,
+        part_eq_glue: Option<Glue<fn(&Self, &Self) -> bool>>,
```
Comment (andrewvyrros) [inline rust/src/ffi/any.rs:44]: Still a couple of accidental renames here.

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**